pennylane/boolean_fn.py

```diff
@@ -82,13 +83,16 @@ def __init__(self, fn):
         functools.update_wrapper(self, fn)
 
     def __and__(self, other):
-        return BooleanFn(lambda obj: self.fn(obj) and other.fn(obj))
+        return BooleanFn(lambda *args, **kwargs: self(*args, **kwargs) and other(*args, **kwargs))
```

Comment:
```suggestion
        return BooleanFn(lambda *args, **kwargs: self(*args, **kwargs) & other(*args, **kwargs))
```
Not a serious suggestion at this point, but I'm curious on whether we should use `and` or bitwise `&` here, since my understanding is that the `__and__` magic method is for bitwise `&`.

Reply:
I was wondering the same when I saw we were using `and` [here](https://github.com/PennyLaneAI/pennylane/blob/9f7e8ba8f41580fc3efe88bf88f0cc05c04a66ab/pennylane/boolean_fn.py#L85). 🤔